File: go/prime-factors/prime_factors.go
package prime

func Factors(n int64) []int64 {
	factors := []int64{}
	for i := int64(2); i <= n; i++ {
        for n % i == 0 {
            factors = append(factors, i)
            n = n / i
        }
    	if n == 1 {
            break
        }
    }
	return factors
}

File: clojure/prime-factors/src/prime_factors.clj
(ns prime-factors)

(defn of [number]
  (loop [number number
         candidate 2
         factors []]
    (cond
      (= number 1) factors
      (zero? (mod number candidate))
        (recur (/ number candidate) candidate (conj factors candidate))
      :else
        (recur number (inc candidate) factors))))
